test/unitTests/build/reportGenerator.test.ts

```diff
@@ -0,0 +1,167 @@
+import assert from 'assert';
+import fs from 'fs-extra';
+import path from 'path';
+import vscode from 'vscode';
+import * as utils from '../../../src/utils/utils';
+import { EventStream } from '../../../src/common/eventStream';
+import { SinonSandbox, createSandbox, SinonStub } from 'sinon';
+import TestEventBus from '../../utils/testEventBus';
+import { DiagnosticController } from '../../../src/build/diagnosticController';
+import { Uri, Diagnostic, Range } from 'vscode';
+import { visualizeBuildReport } from '../../../src/build/reportGenerator';
+import { BuildProgress } from '../../../src/common/loggingEvents';
+
+describe("ReportGenerator", () => {
+    const testRepositoryPath = path.resolve(__dirname, "fakedFolder");
+    const fakedErrorLog = `{"message_severity":"info","log_item_type":"user","code":"author-missing","message":"Missing required attribute: 'author'. Add the current author's GitHub ID.","file":"index.md","line":1,"end_line":1,"column":1,"end_column":1,"date_time":"2020-03-04T08:43:12.3284386Z"}\n`

```

Comment:
Regarding the fake test data,
is it possible to use [autofixture](https://www.npmjs.com/package/autofixture#creating-a-fixtures-file) to shorten the test code?

Reply:
actually, IMO, for this case, the test data are all string value instead of several object with the same structure, `autofixture` may be not helpful to shorten the code. or Can you provide some sample if have some idea about that?

---
In reply to: [394073088](https://github.com/docascode/vscode-docs-build/pull/94#discussion_r394073088) [](ancestors = 394073088)